### PR TITLE
Reset service worker files with new create-react-app defaults

### PR DESCRIPTION
**Pull request by CarsonHerness**
create-react-app has a new template for service workers. This PR essentially is a reset on the service worker that enables the PWA, and it should address at least some of the PWA bugs we're facing.

### CODE REVIEW

File: public/manifest.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
       "sizes": "512x512"
     }
   ],
-  "start_url": "",
+  "start_url": ".",
```
Comment (akampen) [inline public/manifest.json:21]: Just wondering, what does this accomplish?

File: src/service-worker.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
+/// <reference lib="webworker" />
```
Comment (akampen) [inline src/service-worker.ts:1]: Is this supposed to be commented out?

Comment (akampen) [reply src/service-worker.ts:1]: Is this file auto-generated?

File: src/serviceWorkerRegistration.ts [hunk 3]
```diff
@@ -14,34 +14,68 @@ type Config = {
   onUpdate?: (registration: ServiceWorkerRegistration) => void;
 };
 
-function registerValidSW(swUrl: string, config?: Config): void {
+export function register(config?: Config): void {
```
Comment (akampen) [inline src/serviceWorkerRegistration.ts:17]: Add docstring

File: src/serviceWorkerRegistration.ts [hunk 4]
```diff
@@ -14,34 +14,68 @@ type Config = {
...
+        });
+      } else {
+        // Is not localhost. Just register service worker
+        registerValidServiceWorker(serviceWorkerUrl, config);
+      }
+    });
+  }
+}
+
+function registerValidServiceWorker(serviceWorkerUrl: string, config?: Config) {
```
Comment (akampen) [inline src/serviceWorkerRegistration.ts:51]: Add doc string

File: src/serviceWorkerRegistration.ts [hunk 5]
```diff
@@ -14,34 +14,68 @@ type Config = {
...
 
               // Execute callback
               if (config && config.onUpdate) {
                 config.onUpdate(registration);
               }
             } else {
-              // Everything has been pre-cached for offline use
+              // At this point, everything has been precached.
+              // It's the perfect time to display a
+              // "Content is cached for offline use." message.
```
Comment (akampen) [inline src/serviceWorkerRegistration.ts:78]: Delete

File: src/serviceWorkerRegistration.ts [hunk 6]
```diff
@@ -58,15 +92,14 @@ function registerValidSW(swUrl: string, config?: Config): void {
     });
 }
 
-function checkValidServiceWorker(swUrl: string, config?: Config): void {
+function checkValidServiceWorker(serviceWorkerUrl: string, config?: Config) {
```
Comment (akampen) [inline src/serviceWorkerRegistration.ts:95]: Doc string

File: src/serviceWorkerRegistration.ts [hunk 7]
```diff
@@ -91,40 +124,6 @@ function checkValidServiceWorker(swUrl: string, config?: Config): void {
...
-        });
-      } else {
-        // Is not localhost. Just register service worker
-        registerValidSW(swUrl, config);
-      }
-    });
-  }
-}
-
 export function unregister(): void {
```
Comment (akampen) [inline src/serviceWorkerRegistration.ts:127]: Doc string

**Review by akampen — APPROVED** [review]

File: public/manifest.json (hunk 1 above)
Comment (CarsonHerness) [reply public/manifest.json:21]: It was the default used by other manifest files, but I think we do actually want the old value here, so I changed it back.

File: src/service-worker.ts (hunk 2 above)
Comment (CarsonHerness) [reply src/service-worker.ts:1]: This is actually the syntax for Triple Slash Directives (see [documentation](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html)). This was autogenerated, so I don't have the strongest understanding about what this does, but my current understanding is that it's used by the TypeScript compiler.

**Pull request merged**